pyproject.toml

```diff
@@ -15,15 +15,33 @@ repository = "https://github.com/MPI-IS/h5darkframes"
 
 
 [tool.poetry.dependencies]
-python = ">=3.8,<4"
+python = "^3.10"
+# Normal dependencies:
+# Installable with both poetry and pip.
 numpy = "^1.23.4"
 alive-progress = "^2.4.1"
 h5py = "^3.7.0"
 toml = "^0.10.2"
 opencv-python = "^4.6.0.66"
-pytest = "^7.1.3"
 rich = "^12.6.0"
 
+[tool.poetry.group.dev]
+optional = true
+
+[tool.poetry.group.dev.dependencies]
+# Dependencies for development:
+# Only installable with poetry (NOT pip).
+isort = "^5.13.2"
+mypy = "^1.10.0"
+
+[tool.poetry.group.test]
+optional = true
+
+[tool.poetry.group.test.dependencies]
+# Dependencies for testing:
+# Only installable with poetry (NOT pip).
+pytest = "^8.2.0"
```

Comment:
Use version 8.x so that version solving via `poetry` works when setting up this project as a submodule for `nightskycam` (which uses  the range `"^8.1.1"`). 🙂